9.2/root/usr/share/container-scripts/postgresql/common.sh

```diff
@@ -1,6 +1,21 @@
 # Configuration settings.
 export POSTGRESQL_MAX_CONNECTIONS=${POSTGRESQL_MAX_CONNECTIONS:-100}
-export POSTGRESQL_SHARED_BUFFERS=${POSTGRESQL_SHARED_BUFFERS:-32MB}
+
+# Perform auto-tuning based on the container cgroups limits (only when the
+# limits are set).
+# Users can still override this by setting the POSTGRESQL_SHARED_BUFFERS
+# variable.
+if [[ "${MAX_MEMORY_LIMIT_IN_BYTES}" != "${MEMORY_LIMIT_IN_BYTES}" ]]; then
+    # Use 1/4 of given memory for shared buffers
+    shared_buffers_computed="$(python -c "print(int((${MEMORY_LIMIT_IN_BYTES}/(1024*1024))*0.25))")M"
+    # Setting effective_cache_size to 1/2 of total memory would be a normal conservative setting,
+    effective_cache="$(python -c "print(int((${MEMORY_LIMIT_IN_BYTES}/(1024*1024))*0.5))")M"
+    export POSTGRESQL_SHARED_BUFFERS=${POSTGRESQL_SHARED_BUFFERS:-$shared_buffers_computed}
+    export POSTGRESQL_EFFECTIVE_CACHE_SIZE=${POSTGRESQL_EFFECTIVE_CACHE_SIZE:-$effective_cache}
```

Comment:
why the fallback to defaults here?  you know you have a value since you just computed it.